torch/_dynamo/variables/nn_module.py

```diff
@@ -216,7 +216,8 @@ def record_nn_module_stack():
                 # unroll Sequential()
                 assert not kwargs
                 (arg,) = args
-                for child_name, submod in mod.named_children():
+                # TODO: Use named_children when it supports remove_duplicate.
+                for child_name, submod in mod._modules.items():
```

Comment:
"Why do you need to access that internal attribute instead of using the public APIs?" - @albanD

Reply:
Please read the whole thread, the story behind this decision is clear addressed.